src/main/java/uk/gov/justice/digital/hmpps/whereabouts/config/JwkClient.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.hmpps.whereabouts.config;
+
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyConverter;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.security.Key;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+
+@Component
+public class JwkClient {
+
+    private RestTemplate restTemplate;
+    private String keyId;
+
+    public JwkClient(@Qualifier("oauthApiHealthRestTemplate") final RestTemplate oauthApiHealthRestTemplate,
+                     @Value("${jwt.jwk.key.id}") final String keyId) {
```

Comment:
Spring security takes the key id from the JWT header and ideally we would do the same here.  Unfortunately Spring uses a package access class JwtHeaderConverter which is not available to us so we'd have to create our own version...

The benefit of taking the kid from the JWT header is that there is no change required in the client when we add a new JWK set hence we can rotate keys by simply adding them to the Oauth server and restarting.